rest-api/dao/organization_hierarchy_sync_dao.py

```diff
@@ -0,0 +1,385 @@
+import os
+import logging
+import googlemaps
+import lib_fhir.fhirclient_3_0_0.models.organization
+
+from lib_fhir.fhirclient_3_0_0.models.fhirabstractbase import FHIRValidationError
+from werkzeug.exceptions import BadRequest
+from dao.base_dao import BaseDao
+from model.hpo import HPO
+from model.organization import Organization
+from model.site import Site
+from model.site_enums import ObsoleteStatus, SiteStatus, EnrollingStatus, DigitalSchedulingStatus
+from dao.hpo_dao import HPODao
+from participant_enums import OrganizationType
+from dao.organization_dao import OrganizationDao
+from dao.site_dao import SiteDao
+from dateutil.parser import parse
+
+_FHIR_SYSTEM_PREFIX = 'http://all-of-us.org/fhir/sites/'
+
+
+class OrganizationHierarchySyncDao(BaseDao):
+  def __init__(self):
+    super(OrganizationHierarchySyncDao, self).__init__(HPO)
+    self.hpo_dao = HPODao()
+    self.organization_dao = OrganizationDao()
+    self.site_dao = SiteDao()
+
+  def from_client_json(self, resource_json, id_=None, expected_version=None, client_id=None):  # pylint: disable=unused-argument
+    try:
+      fhir_org = lib_fhir.fhirclient_3_0_0.models.organization.Organization(resource_json)
+    except FHIRValidationError:
+      raise BadRequest('Invalid FHIR format in payload data.')
+
+    if not fhir_org.meta or not fhir_org.meta.versionId:
+      raise BadRequest('No versionId info found in payload data.')
+    try:
+      fhir_org.version = int(fhir_org.meta.versionId)
+    except ValueError:
+      raise BadRequest('Invalid versionId in payload data.')
+
+    return fhir_org
+
+  def to_client_json(self, hierarchy_org_obj):
+    return hierarchy_org_obj.as_json()
+
+  def get_etag(self, id_, pid):  # pylint: disable=unused-argument
+    return None
+
+  def update(self, hierarchy_org_obj):
+    obj_type = self._get_type(hierarchy_org_obj)
+
+    operation_funcs = {
+      'AWARDEE': self._update_awardee,
+      'ORGANIZATION': self._update_organization,
+      'SITE': self._update_site
+    }
+
+    if obj_type not in operation_funcs:
+      raise BadRequest('No awardee-type info found in payload data.')
```

Comment:
I think this wording could become confusing, maybe something without explicitly mentioning one of (awardee, organization, site)

Reply:
'awardee-type' is the field name in their payload.